api/src/main/java/org/openmrs/module/bedmanagement/BedManagementActivator.java

```diff
@@ -13,16 +13,19 @@
  */
 package org.openmrs.module.bedmanagement;
 
-import org.openmrs.module.ModuleActivator;
-
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.BaseModuleActivator;
+import org.openmrs.module.ModuleActivator;
 import org.openmrs.module.bedmanagement.constants.BedManagementProperties;
 
+import java.util.List;
+
 /**
  * This class contains the logic that is run every time this module is either started or stopped.
  */
-public class BedManagementActivator implements ModuleActivator {
+public class BedManagementActivator extends BaseModuleActivator {
 	
```

Comment:
I followed the same approach here that I introduced a few months ago with the Bahmni appointments module, creating a n interface that allows plugging in spring components to execute during the activator lifecycle.  This allows us to wire in beans from conditional resources, thus enabling certain behavior when atom-feed is installed that is does not occur otherwise.